### DISCUSSION TITLE
switchTo is throwing Error Message: TypeError: Cannot read property 'concat' of null

### DISCUSSION THREAD

**Discussion by nithin936sam** [getgauge/taiko#2255] [General]
switchTo is throwing Error Message: TypeError: Cannot read property 'concat' of null

Specification: specs\test.spec:16
        Error Message: TypeError: Cannot read property 'concat' of null
        Stacktrace:
        TypeError: Cannot read property 'concat' of null
            at isMatchingRegex (node_modules\taiko\lib\handlers\targetHandler.js:55:42)
            at Object.getCriTargets (node_modules\taiko\lib\handlers\targetHandler.js:219:7)
            at async module.exports.switchTo (node_modules\taiko\lib\taiko.js:222:21)
            at async module.exports.<computed> (node_modules\taiko\lib\taiko.js:2612:14)
            at async Object.<anonymous> (tests\operatorStep.js:161:5)

Gauge version: 1.1.8
Commit Hash: b1501f4
Plugins
-------
html-report (4.0.12)
js (2.3.16)

**Comment by saikrishna321** [comment]
Can you please provide a sample code snippet to reproduce this problem?

**Reply by nithin936sam** [reply]
1. await click("button on parent window",{ navigationTimeout: 60000, force: true });
2. await switchTo("newly opened tab which is a user authentication page");
3. await write("enter username",{ navigationTimeout: 30000, force: true }));
4. await write("enter password",{ navigationTimeout: 30000, force: true }));
5. await click("SIGN IN", { navigationTimeout: 30000, force: true });

Concat issue is observed in 2nd step

Earlier this code has passed 2nd step and it was able to switch to the newly opened tab and authorize user. However when its not able to switch back focus to parent window after 5th step because if its successful authorization then newly opened tab closes and I want my focus again on parent window which is not happening and its throwing navigation timeout error

**Reply by zabil** [reply]
From the code it looks like and error parsing the url's
```
const urlHrefPath = parsedUrl.protocol.concat('//').concat(trimCharLeft(host, '/'));
```

Can you be specific about the arguments passed to 2? If the info is sensitive, please share something that is similar in structure.

**Reply by nithin936sam** [reply]
I am switching tabs using Title. For example on executing 1st step it opens a new tab with Title ABC, I am using await switchTo(/ABC/)

Also my main concern is after step5, where the newly opened tab is closed after successful authorization and script is throwing navigation timeout error

As a workaround for step2 I am using keyboard controls to shift focus on other tab

**Reply by nithin936sam** [reply]
@saikrishna321 @zabil --> Could you please help here as I don't see any workaround after performing step5

**Reply by zabil** [reply]
I think that issue was fixed in the newer versions of Taiko. Can you share what version of node js and taiko you are on?

**Reply by SarahTaaherBonna** [reply]
Hi, I am facing the same issue. My node js version is v14.17.1 and taiko version is 1.2.7. Would appreciate the help. Thank you.
